drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -372,16 +380,66 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 	}
 
 	if (i == clk_values) {
-		dev_err(bus->dev, "%s: could not find clock value for bandwidth %d\n",
+multilane:
+		dev_dbg(bus->dev, "%s: could not find clock value for bandwidth %d, checking multi-lane\n",
 			__func__, bus->params.bandwidth);
-		return -EINVAL;
+
+		m_rt = list_last_entry(&bus->m_rt_list, struct sdw_master_runtime, bus_node);
+		list_for_each_entry(s_rt, &m_rt->slave_rt_list, m_rt_node) {
+			slave_prop = &s_rt->slave->prop;
+
+			/* Find a non-zero manager lane */
+			for (i = 1; i < SDW_MAX_LANES; i++) {
+				if (!slave_prop->lane_maps[i])
+					continue;
+
+				required_bandwidth = 0;
+				list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+					required_bandwidth += m_rt->stream->params.rate *
+						hweight32(m_p_rt->ch_mask) *
+						m_rt->stream->params.bps;
+				}
+				if (required_bandwidth <= curr_dr_freq - bus->lane_used_bandwidth[i]) {
+					s_lane = i;
+					m_lane = slave_prop->lane_maps[i];
+					dev_dbg(&s_rt->slave->dev,
+						"M lane %d P lane %d can be used\n",
+						m_lane, s_lane);
+					bus->lane_used_bandwidth[i] += required_bandwidth;
+					bus->params.bandwidth -= required_bandwidth;
+					/*
+					 * Use non-zero manager lane, subtract the lane 0
+					 * bandwidth that is already calculated
+					 * */
+					use_multi_lane = true;
+					break;
+				}
+			}
+
+			list_for_each_entry(s_p_rt, &s_rt->port_list, port_node) {
+				s_p_rt->lane = s_lane;
+			}
+			list_for_each_entry(m_p_rt, &m_rt->port_list, port_node) {
+				m_p_rt->lane = m_lane;
+			}
```

Comment:
that last loop should not be dependent on the number of peripherals connected. If there are N peripherals connected, this will be assigned N times, and possible one iteration overrides the previous settings.

It's really the same comment as above, the lane selection has to be the intersection of what all devices support, and the bandwidth counted once.

Reply:
so in the new iteration you verify the intersection but the bandwidth is still counted multiple times.

Reply:
> so in the new code update you verify the intersection but the bandwidth is still counted multiple times.

Does the `bandwidth` here means `required_bandwidth`? It will be reset when we test each lane. So, it is not counted multiple times. `bus->params.bandwidth` is still used to select row and column. And if multi-lane is used, move the bandwidth from `bus->params.bandwidth` to bus->lane_used_bandwidth[i].